numpyro/mcmc.py

```diff
@@ -212,7 +212,7 @@ def init_kernel(init_params,
         step_size = lax.convert_element_type(step_size, xla_bridge.canonicalize_dtype(np.float64))
         nonlocal momentum_generator, wa_update, trajectory_len, max_treedepth, wa_steps
         wa_steps = num_warmup
-        trajectory_len = float(trajectory_length)
+        trajectory_len = trajectory_length
```

Comment:
step_size is float, so we don't need to cast trajectory_len to float. Historically, I guess these casting are necessary because we use float division at `num_steps = trajectory_length / step_size`

Reply:
Yes, I think we needed this earlier if a user passed ints for both `step_size` and `trajectory_length` but with the `convert_element_type` in the first line, we don't need it anymore.